metadata-ingestion/tests/unit/serde/test_codegen.py

```diff
@@ -170,3 +185,136 @@ def test_urn_types() -> None:
+                should_log_warning=False,
+            ),
+            id="basic_default",
+        ),
+        pytest.param(
+            AuditStampFromTokenTestParams(
+                auth_token=None,
+                fallback_actor=CorpUserUrn("custom-fallback"),
+                fallback_timestamp=None,
+                expected_actor="urn:li:corpuser:custom-fallback",
+                should_log_warning=False,
+            ),
+            id="custom_fallback_user",
+        ),
+        pytest.param(
+            AuditStampFromTokenTestParams(
+                auth_token=None,
+                fallback_actor=CorpGroupUrn("data-engineers"),
+                fallback_timestamp=None,
+                expected_actor="urn:li:corpGroup:data-engineers",
+                should_log_warning=False,
+            ),
+            id="custom_fallback_group",
+        ),
+        pytest.param(
+            AuditStampFromTokenTestParams(
+                auth_token=jwt.encode(
+                    {"actorId": "test-user", "type": "PERSONAL"},
+                    "secret",
+                    algorithm="HS256",
+                ),
+                fallback_actor=None,
+                fallback_timestamp=None,
+                expected_actor="urn:li:corpuser:test-user",
+                should_log_warning=False,
+            ),
+            id="valid_jwt",
+        ),
+        pytest.param(
+            AuditStampFromTokenTestParams(
+                auth_token="invalid-token",
+                fallback_actor=None,
+                fallback_timestamp=None,
+                expected_actor="urn:li:corpuser:__ingestion",
+                should_log_warning=True,
+            ),
+            id="invalid_jwt",
+        ),
+        pytest.param(
+            AuditStampFromTokenTestParams(
+                auth_token=jwt.encode(
+                    {"type": "PERSONAL"}, "secret", algorithm="HS256"
+                ),
+                fallback_actor=None,
+                fallback_timestamp=None,
+                expected_actor="urn:li:corpuser:__ingestion",
+                should_log_warning=True,
+            ),
+            id="jwt_missing_actor",
+        ),
+        pytest.param(
+            AuditStampFromTokenTestParams(
+                auth_token=jwt.encode(
+                    {"actorId": "service-account", "type": "SERVICE"},
+                    "secret",
+                    algorithm="HS256",
+                ),
+                fallback_actor=None,
+                fallback_timestamp=None,
+                expected_actor="urn:li:corpuser:__ingestion",
+                should_log_warning=True,
+            ),
+            id="jwt_non_personal",
+        ),
+        pytest.param(
+            AuditStampFromTokenTestParams(
+                auth_token=jwt.encode(
+                    {"actorId": "token-user", "type": "PERSONAL"},
+                    "secret",
+                    algorithm="HS256",
```

Comment:
can we have at least one test that uses a real datahub-generated (expired) token instead of creating one using jwt.encode